### PR TITLE
Revamped help command and extended d.pys command framework

### PR DESCRIPTION
**Pull request by Jay-Madden**
Also expanded the paginators ability to accept a list of Embeds

Closes #228 and #127

### CODE REVIEW

**Comment by Jay-Madden** [comment]
lgtm

File: bot/cogs/assignable_roles_cog.py [hunk 1]
```diff
@@ -197,6 +198,9 @@ async def remove_role(self, ctx, role: discord.Role):
 
     @roles.command(pass_context= True, aliases= ['create'])
     @commands.has_guild_permissions(administrator = True)
+    @ext.long_help('Command to add a role as assignable in the current guild')
+    @ext.short_help('Marks a role as user assignable')
+    @ext.example('roles add #SomeExampleRole')
```
Comment (smathep) [inline bot/cogs/assignable_roles_cog.py:203]: Shouldn't it use an @ instead of # for roles?

Comment (smathep) [reply bot/cogs/assignable_roles_cog.py:203]: same for line 216

Comment (Jay-Madden) [reply bot/cogs/assignable_roles_cog.py:203]: ope yes, i did this after the designated channel helps and just mindlessly did it, good catch

File: bot/cogs/example_cog.py [hunk 2]
```diff
@@ -27,10 +31,17 @@ def __init__(self, bot):
         self.bot = bot
         self._last_member = None
 
-    #To createa a command you decorate an async method with commands.command
```
Comment (smathep) [inline bot/cogs/example_cog.py:30]: The typo with create has been bugging me since I started writing the translate cog lol. It needs to be fixed

**Pull request merged**